circuits/cpp/src/aztec3/circuits/hash.hpp

```diff
@@ -72,17 +74,23 @@ typename NCT::address compute_contract_address_from_partial(Point<NCT> const& po
     return { NCT::hash(inputs, aztec3::GeneratorIndex::CONTRACT_ADDRESS) };
 }
 
-template <typename NCT> typename NCT::address compute_contract_address(Point<NCT> const& point,
-                                                                       typename NCT::fr const& contract_address_salt,
-                                                                       typename NCT::fr const& function_tree_root,
-                                                                       typename NCT::fr const& constructor_hash)
+template <typename NCT> typename aztec3::circuits::abis::CompleteAddress<NCT> compute_complete_address(
+    Point<NCT> const& point,
+    typename NCT::fr const& contract_address_salt,
+    typename NCT::fr const& function_tree_root,
+    typename NCT::fr const& constructor_hash)
 {
     using fr = typename NCT::fr;
 
     const fr partial_address =
         compute_partial_address<NCT>(contract_address_salt, function_tree_root, constructor_hash);
 
-    return compute_contract_address_from_partial(point, partial_address);
+    typename aztec3::circuits::abis::CompleteAddress<NCT> complete_address;
```

Comment:
I feel like this block of code would be best in a static member of CompleteAddress.